### PR TITLE
(BSR) chore(tests): remove wait for expect for some tests - part 1

### PR DESCRIPTION
**Pull request by Lucasbeneston**
## Checklist

I have:

- [x] Made sure my feature is working on the relevant real / virtual devices (native and web).
- [x] Written **unit tests** native (and web when implementation is different) for my feature.
- [ ] Added a **screenshot** for UI tickets or deleted the screenshot section if no UI change
- [ ] If my PR is a bugfix, I add the link of the "résolution de problème sur le bug" [on Notion][1]

### CODE REVIEW

**Review by aliraiki — APPROVED** [review]

File: src/features/auth/pages/signup/SignupForm.native.test.tsx [hunk 1]
```diff
@@ -65,113 +65,134 @@ describe('<SignupForm />', () => {
...
     const props = {
       ...defaultProps,
       route: { ...defaultProps.route, params: { preventCancellation: true } },
     }
-    const { queryByTestId } = render(<SignupForm {...props} />)
-    const icon = queryByTestId('Abandonner l’inscription')
+    render(<SignupForm {...props} />)
+
+    const icon = screen.queryByTestId('Abandonner l’inscription')
     expect(icon).toBeFalsy()
```
Comment (marineshaw) [inline src/features/auth/pages/signup/SignupForm.native.test.tsx:83]:
```suggestion
    expect(screen.queryByTestId('Abandonner l’inscription')).toBeFalsy()
```

File: src/features/auth/pages/signup/SignupForm.native.test.tsx [hunk 2]
```diff
@@ -65,113 +65,134 @@ describe('<SignupForm />', () => {
...
   it('should call logContinueSetPassword when clicking on next step from SetPassword', async () => {
-    const { getByTestId, getByText } = render(<SignupForm {...defaultProps} />)
-    fireEvent.press(getByTestId('goToNextStep'))
-    getByText('SetPassword')
-    fireEvent.press(getByTestId('goToNextStep'))
+    render(<SignupForm {...defaultProps} />)
+
+    fireEvent.press(screen.getByTestId('goToNextStep'))
+
+    screen.getByText('SetPassword')
```
Comment (marineshaw) [inline src/features/auth/pages/signup/SignupForm.native.test.tsx:144]: Et lui ? 🧐

File: src/features/auth/pages/signup/SignupForm.native.test.tsx [hunk 3]
```diff
@@ -65,113 +65,134 @@ describe('<SignupForm />', () => {
...
-    fireEvent.press(getByTestId('goToNextStep'))
+    render(<SignupForm {...defaultProps} />)
 
-    await waitForExpect(() => {
+    fireEvent.press(screen.getByTestId('goToNextStep'))
+
+    screen.getByText('SetPassword')
+    fireEvent.press(screen.getByTestId('Revenir en arrière'))
+
+    screen.getByText('SetEmail')
```
Comment (marineshaw) [inline src/features/auth/pages/signup/SignupForm.native.test.tsx:160]: idem here

File: src/features/auth/pages/signup/SignupForm.native.test.tsx [hunk 4]
```diff
@@ -65,113 +65,134 @@ describe('<SignupForm />', () => {
...
-    fireEvent.press(getByTestId('goToNextStep'))
-    fireEvent.press(getByTestId('goToNextStep'))
-    getByText('SetBirthday')
-    fireEvent.press(getByTestId('goToNextStep'))
+    render(<SignupForm {...defaultProps} />)
+
+    fireEvent.press(screen.getByTestId('goToNextStep'))
+    fireEvent.press(screen.getByTestId('goToNextStep'))
+
+    screen.getByText('SetBirthday')
```
Comment (marineshaw) [inline src/features/auth/pages/signup/SignupForm.native.test.tsx:175]: 👋

File: src/features/auth/pages/signup/SignupForm.native.test.tsx [hunk 5]
```diff
@@ -65,113 +65,134 @@ describe('<SignupForm />', () => {
...
-    fireEvent.press(getByTestId('goToNextStep'))
-    getByText('AcceptCgu')
-    fireEvent.press(getByTestId('goToNextStep'))
+    render(<SignupForm {...defaultProps} />)
+
+    fireEvent.press(screen.getByTestId('goToNextStep'))
+    fireEvent.press(screen.getByTestId('goToNextStep'))
+    fireEvent.press(screen.getByTestId('goToNextStep'))
+
+    screen.getByText('AcceptCgu')
```
Comment (marineshaw) [inline src/features/auth/pages/signup/SignupForm.native.test.tsx:192]: Last one 🤠

File: src/features/identityCheck/pages/confirmation/IdentityCheckHonor.native.test.tsx [hunk 6]
```diff
@@ -49,10 +48,9 @@ describe('<IdentityCheckHonor/>', () => {
     const button = getByText('Valider et continuer')
```
Comment (marineshaw) [inline src/features/identityCheck/pages/confirmation/IdentityCheckHonor.native.test.tsx:48]: Tu pourrais utiliser screen ici aussi 🤓

File: src/features/identityCheck/pages/profile/SetCity.native.test.tsx [hunk 7]
```diff
@@ -50,71 +49,62 @@ describe('<SetCity/>', () => {
...
-      expect(mockNavigateToNextScreen).toBeCalledTimes(1)
-      expect(mockDispatch).toHaveBeenNthCalledWith(1, {
-        type: 'SET_CITY',
-        payload: {
-          code: city.code,
-          name: city.nom,
-          postalCode: POSTAL_CODE,
-        },
-      })
+    await screen.findByText(city.nom)
```
Comment (marineshaw) [inline src/features/identityCheck/pages/profile/SetCity.native.test.tsx:71]: Trop bien comme correction ça, c'était super moche avant !!

**Pull request merged**